### DISCUSSION TITLE
Grid line issue

### DISCUSSION THREAD

**Discussion by Remsdel** [mtex-toolbox/mtex#1799] [Ask Anything]
When reading a ctf file in MTEX I very often get non-indexed vertical and horizontal lines (see picture attached). I understood that this is a known issue caused by a rounding error in the x-y coordinates of data points which occurs when they are converted to a regular grid by MTEX.  It is reported in some MTEX manual that this can be fixed only by renumbering the XY coordinates in the text file to be integer multiples of the grid spacing listed in the header of the text file.
However, because this issue that is very often present in my maps, I wonder if somebody has implemented a script that can automatically correct the data to avoid those lines.
![Non_indexed_map_grid_line_issue](https://user-images.githubusercontent.com/78489177/233641380-580ea358-7811-4079-aee5-fed43e4cc37e.png)

**Comment by ralfHielscher** [comment]
have you already tried
```matlab
ebsd = ebsd.gridify
```
Ralf.

**Comment by Remsdel** [comment]
Sorry I completely forgot to mention that I work on gridified data.
I just checked actually and the issue seems to come from the gridification itself (which is not surprising) as the lines do not appear when plotting the non gridified ebsd data. I would be nice if the gridify function could handle those rounding errors directly.

Remi

**Reply by ralfHielscher** [reply]
Can you provide the data or some piece of it that show the problem?

Ralf.

**Comment by Remsdel** [comment]
Here is the ebsd.mat (zipped) file that shows the issues when gridified 

[ebsd.zip](https://github.com/mtex-toolbox/mtex/files/11296347/ebsd.zip)

**Reply by ralfHielscher** [reply]
Hi Remsdel,

thank you for the data. I could reproduce the problem. The issue is with the unitCell computation.
In your specify case you can do
```
ebsd.unitCell = [
    0.1130    0.1130
   -0.1130    0.1130
   -0.1130   -0.1130
    0.1130   -0.1130
];
```
We will shortly release a version of MTEX which fixes this problem.
Thank you for reporting.

Ralf.

**Reply by ralfHielscher** [reply]
The issue has been resolved in 0d3fce6

**Reply by Remsdel** [reply]
The same problem has re-surfaced when using MTEX 6.0.0. Were the modifictions mentioned above carried on to the version 6.00?
